include/tscpp/api/Plugin.h

```diff
@@ -58,6 +58,7 @@ class Plugin : noncopyable
     HOOK_READ_REQUEST_HEADERS,  /**< This hook will be fired after the request is read. */
     HOOK_READ_CACHE_HEADERS,    /**< This hook will be fired after the CACHE hdrs. */
     HOOK_CACHE_LOOKUP_COMPLETE, /**< This hook will be fired after cache lookup complete. */
+    HOOK_TXN_CLOSE,             /**< This hook will be fired after send response headers. */
```

Comment:
I think it would be good to add a "only for TransactionPlugins::registerHood()!" comment here.

Reply:
I think you should add 'TSAssert(hook_type != Plugin::HOOK_TXN_CLOSE);' after this line:
https://github.com/apache/trafficserver/blob/3376d438b4a6410187e1ddedd87d2e89279ec196/src/tscpp/api/GlobalPlugin.cc#L89

Reply:
Ack. Makes sense.